src/lib/transaction_snark_scan_state/transaction_snark_scan_state.ml

```diff
@@ -360,24 +360,22 @@ let create_expected_statement ~constraint_constants
   ; sok_digest = ()
   }
 
-let completed_work_to_scanable_work ~proof_cache_db (job : job)
-    (fee, current_proof, prover) : Ledger_proof_with_sok_message.t Or_error.t =
-  let sok_digest = Ledger_proof.sok_digest current_proof
-  and proof = Ledger_proof.underlying_proof current_proof in
+let completed_work_to_scanable_work (job : job) (fee, current_proof, prover) :
+    Ledger_proof_with_sok_message.t Or_error.t =
+  let sok_digest = Ledger_proof.Cached.sok_digest current_proof
+  and proof = Ledger_proof.Cached.underlying_proof current_proof in
   match job with
   | Base { statement; _ } ->
       let ledger_proof =
-        Ledger_proof.create ~statement ~sok_digest ~proof
-        |> Ledger_proof.Cached.write_proof_to_disk ~proof_cache_db
+        Ledger_proof.Cached.create ~statement ~sok_digest ~proof
```

Comment:
This is just reconstructing the input `current_proof`, you can just pass it through.